### PR TITLE
[Moveit_servo] Move and rename jogger

### PR DESCRIPTION
**Pull request by AdamPettinger**
### Description

I moved and renamed the jogger: `moveit_experimental/moveit_jog_arm` -> `moveit_ros/moveit_servo`. In addition to the file paths and names, I renamed some of the internals (namespaces, project/package names, JogArm interface class -> Servo interface class, etc). This PR overall looks like a ton of changes, but I tried to break it down so each commit is a logical step that only modifies file names, or changed the files insides so that no commit in itself had a tremendous amount of diffs. 

Also kind of weird to test, but I copied the new `moveit_servo` package to a catkin workspace and built it there successfully. I am currently working on getting this to build with ROS2, so the COLCON_IGNORE will stay for a bit longer

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/moveit_servo/CMakeLists.txt [hunk 1]
```diff
@@ -84,11 +84,11 @@ target_link_libraries(cpp_interface_example
 ## ROS message-based node ##
 ############################
 
-add_executable(jog_server
-  src/jog_server.cpp
+add_executable(servo_server
```
Comment (AndyZe) [inline moveit_ros/moveit_servo/CMakeLists.txt:87]: I guess servo_server is a bit strange to say, but I'm OK with it

**Comment by AdamPettinger** [comment]
"jog" does still exist in quite a lot of places. There are definitely some places that **need** to be changed in the tests (e.g. `def test_jog_arm_halt_msg(node)`), and I will make those changes

Then throughout the package there are some potential changes that are less obvious. Some examples:
- Are we going to refer to it as joint jogging and Cartesian jogging, or joint servoing and Cartesian servoing?
- "the jogger" is in a lot of comments, what replaces that? 
- Changing JogCalcs -> ServoCalcs? Discussed offline and I think making this change is good
- Parameters in the examples: keep as `delta_jog_cmds`, or replace with what? `delta_servo_cmds`?

**Comment by rhaschke** [comment]
Didn't we want to rename this package in both ROS1 and ROS2?

**Comment by AndyZe** [comment]
> Didn't we want to rename this package in both ROS1 and ROS2?

Sorry I missed the meeting yesterday. After a few weeks of no jog discussion, I thought it was safe to skip. Wrong.

I don't want to give Adam more menial tasks, nor distract him from the main goal. If you feel strongly that it should get renamed in ROS1, I can do that. Do you feel strongly?

**Comment by AndyZe** [comment]
>     * Are we going to refer to it as joint jogging and Cartesian jogging, or joint servoing and Cartesian servoing?

I think we should purge "jog" as much as possible. So, joint servoing and Cartesian servoing.

>     * "the jogger" is in a lot of comments, what replaces that?

"the servo server"? "the servo node"?

>     * Changing JogCalcs -> ServoCalcs? Discussed offline and I think making this change is good

yep, i think so too

>     * Parameters in the examples: keep as `delta_jog_cmds`, or replace with what? `delta_servo_cmds`?

Don't we have `delta_joint_cmds` and `delta_jog_cmds`? I think it should be `delta_joint_cmds` and `delta_cartesian_cmds`

**Comment by AdamPettinger** [comment]
I think these sound good and what I was leaning towards
> So, joint servoing and Cartesian servoing.

> "the servo node"

> I think it should be delta_joint_cmds and delta_cartesian_cmds

But I think I will do `delta_joint_cmds` and `delta_twist_cmds`

**Comment by v4hn** [comment]
> Do you feel strongly?

Talk to Henning about porting from MoveIt1 to MoveIt2. We really want that name to change everywhere, otherwise porting between versions becomes even more of a nightmare!

**Comment by AdamPettinger** [comment]
New commits to address @AndyZe's comments. 

Testing this work in a catkin workspace, so it should work OK in ROS1 if we decide to go that route

**Comment by AdamPettinger** [comment]
Closing this now. With the rename done in ROS1 [https://github.com/ros-planning/moveit/pull/2165](https://github.com/ros-planning/moveit/pull/2165), we don't need to do it here, and should be able to sync those rename changes to moveit2 after #227 is merged

**Pull request closed without merging**